nixos/modules/config/i18n.nix

```diff
@@ -5,15 +5,24 @@
   ...
 }:
 let
+  sanitizeUTF8Capitalization =
+    lang: (lib.replaceStrings [ "utf8" "utf-8" "UTF8" ] [ "UTF-8" "UTF-8" "UTF-8" ] lang);
   aggregatedLocales =
-    (builtins.map
-      (l: (lib.replaceStrings [ "utf8" "utf-8" "UTF8" ] [ "UTF-8" "UTF-8" "UTF-8" ] l) + "/UTF-8")
-      (
-        [ config.i18n.defaultLocale ]
-        ++ (lib.optionals (builtins.isList config.i18n.extraLocales) config.i18n.extraLocales)
-        ++ (lib.attrValues (lib.filterAttrs (n: v: n != "LANGUAGE") config.i18n.extraLocaleSettings))
-      )
-    )
+    [
+      "${config.i18n.defaultLocale}/${config.i18n.defaultCharset}"
+    ]
+    ++ lib.pipe config.i18n.extraLocaleSettings [
+      # TODO: Explain why is this filter added here...
+      (lib.filterAttrs (n: v: n != "LANGUAGE"))
```

Comment:
$LANGUAGE specifies the preferred languages by the user, e.g. `de:en_US`

In my opinion it would be more robust to fail evaluation if `i18n.extraLocaleSettings.X` is set, and X doesn't start with `LC_`.

Reply:
> $LANGUAGE specifies the preferred languages by the user, e.g. `de:en_US`
> 
> In my opinion it would be more robust to fail evaluation if `i18n.extraLocaleSettings.X` is set, and X doesn't start with `LC_`.

I investigated this a bit, and added a 3rd commit addressing and explaining my opinion.